bootloader/wscript

```diff
@@ -628,7 +650,31 @@ def configure(ctx):
         elif ctx.env.DEST_OS == 'hpux' and sysconfig.get_config_var('HAVE_PTHREAD_H'):
             ctx.check_cc(lib='pthread', mandatory=True)
         ctx.check_cc(lib='m', mandatory=True)
-        ctx.check_cc(lib='z', mandatory=True, uselib_store='Z')
+
+        # Opting out of dynamically linked zlib can be done either with the
+        # --static-zlib option or with a PYI_STATIC_ZLIB=1 environment variable.
+        static_zlib = bool(int(os.environ.get("PYI_STATIC_ZLIB", '0')))
+        static_zlib = static_zlib or ctx.options.static_zlib
+        if static_zlib:
+            # This serves as a signal to later on when the C code gets compiled.
+            ctx.env.LIB_Z = None
+        else:
+            ctx.check_cc(lib='z', mandatory=False, uselib_store='Z',
+                         fragment=ZLIB_TEST_FRAGMENT)
+            if not ctx.env.LIB_Z:
+                ctx.fatal(
+                    "The zlib development package is either missing or not "
+                    "linkable to. For security (and marginally better filesize"
+                    "), you should install the zlib-dev or zlib-devel packages"
+                    " with your system package manager and try again. If you "
+                    "can't do this, for example on distributions like OpenWRT "
+                    "which use sstrip on libraries making linking impossible, "
+                    "then either use the --static-zlib option or set the "
+                    "PYI_STATIC_ZLIB=1 environment variable. If you're "
+                    "installing directly with pip, then use the environment "
+                    "variable."
+                )
+
```

Comment:
We should probably avoid contractions in the message.

Reply:
What do you mean by *contractions*?

Reply:
can't, you're